Tips-for-common-editors.md

```diff
@@ -0,0 +1,137 @@
+**A word of warning: Many code editors let you perform git actions from their graphical interfaces instead of through the command line. We strongly recomment that you use the command line instead. Many new contributors run into trouble because their editor's git interface is incorrectly configured.**
+
+## Table of Contents
+
+* [Visual Studio Code](#visual-studio-code)
+* [Sublime](#sublime)
+* [Atom](#atom)
```

Comment:
LGTM! Are we expecting to write a way to run tests easily through the code editor? Is it possible? [IIRC, there exist some plugins to run python-unit test but I'm not sure whether it will work!]

Reply:
@DubeySandeep and I investigated this and found that our Python dependencies setup prevents us from running tests in the editor. In VSCode, for example, we can't specify a custom test executor ([source](https://code.visualstudio.com/docs/python/testing#_enable-a-test-framework)). This means we are stuck using the `unittest` executable, which doesn't make the pythonpath changes we need

Reply:
DubeySandeep and I discussed this, and the next step would be installing our Python dependencies in a normal virtual environment. This would also let us pin all the versions of our dependencies. This has been on the dev workflow team's radar for a while, but it hasn't been a priority. It's tracked in oppia/oppia#11297

Reply:
SGTM, thanks. Maybe talk with @vojtechjelinek too, he set up the current virtualenv solution for Py3 and may have insights.

Reply:
Could you please also update the issue #11297 to say something about making it possible to run tests in the editor once that is fixed? There's no mention of that there.

Reply:
done